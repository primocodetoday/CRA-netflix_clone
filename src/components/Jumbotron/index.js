import React from 'react';
import PropTypes from 'prop-types';
import {
  Item,
  Inner,
  Container,
  Pane,
  Title,
  SubTitle,
  Image,
} from './styles/jumbotron';

const Jumbotron = ({ children, direction }, ...restProps) => {
  return (
    <Item {...restProps} direction={direction}>
      <Inner>{children}</Inner>
    </Item>
  );
};

Jumbotron.Container = function JumbotronContainer({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
};

Jumbotron.Pane = function JumbotronPane({ children, ...restProps }) {
  return <Pane {...restProps}>{children}</Pane>;
};

Jumbotron.Title = function JumbotronTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Jumbotron.SubTitle = function JumboSubtronTitle({ children, ...restProps }) {
  return <SubTitle {...restProps}>{children}</SubTitle>;
};

Jumbotron.Image = function JumbotronImage({ ...restProps }) {
  return <Image {...restProps} />;
};

Jumbotron.propTypes = {
  direction: PropTypes.string,
};

Jumbotron.defaultProps = {
  direction: 'row',
};

export default Jumbotron;
