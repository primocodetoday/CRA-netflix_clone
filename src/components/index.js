export { default as Footer } from 'components/Footer';
export { default as Accordion } from 'components/Accordion';
export { default as Jumbotron } from 'components/Jumbotron';
