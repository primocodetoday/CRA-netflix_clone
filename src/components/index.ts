export { Accordion } from 'components/Accordion/Accordion';
export { Banner } from 'components/Banner/Banner';
export { Card } from 'components/Card/Card';
export { CTAForm } from 'components/CTAForm/CTAForm';
export { Form } from 'components/Form/Form';
export { Footer } from 'components/Footer/Footer';
export { Header } from 'components/Header/Header';
export { Jumbotron } from 'components/Jumbotron/Jumbotron';
export { Loading } from 'components/Loading/Loading';
export { Player } from 'components/Player/Player';
export { Profiles } from 'components/Profiles/Profiles';
