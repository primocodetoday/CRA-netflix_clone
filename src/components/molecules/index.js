export { default as Footer } from 'components/molecules/Footer/Footer';
export { default as Accordion } from 'components/molecules/Accordion/Accordion';
export { default as Jumbotron } from 'components/molecules/Jumbotron/Jumbotron';
export { default as CTAForm } from 'components/molecules/CTAForm/CTAForm';
export { default as Header } from 'components/molecules/Header/Header';
