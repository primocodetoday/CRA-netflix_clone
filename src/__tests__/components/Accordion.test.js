import React from 'react';
import { render } from '@testing-library/react';
import faqsData from 'constants/faqsData.json';
import { Accordion } from 'components';

describe('<Accordion />', () => {
  it('renders the <Accordion /> with populated data', () => {
    const { container, getByText } = render(
      <Accordion>
        <Accordion.Title>Frequently Asked Questions</Accordion.Title>
        {faqsData.map((item) => (
          <Accordion.Item key={item.id}>
            <Accordion.Header>{item.header}</Accordion.Header>
            <Accordion.Body>
              <Accordion.Span>{item.body}</Accordion.Span>
            </Accordion.Body>
          </Accordion.Item>
        ))}
      </Accordion>,
    );

    expect(getByText('Frequently Asked Questions')).toBeTruthy();
    expect(getByText('What is Netflix?')).toBeTruthy();
    expect(getByText('How much does Netflix cost?')).toBeTruthy();
    expect(getByText('Where can I watch?')).toBeTruthy();
    expect(getByText('How do I cancel?')).toBeTruthy();
    expect(getByText('What can I watch on Netflix?')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  // TODO test style props

  // it('opens and closes the <Accordion /> component', () => {
  //   const { component, container, queryByText, queryByTestId } = render(
  //     <Accordion>
  //       {faqsData.map((item) => (
  //         <Accordion.Item key={item.id}>
  //           <Accordion.Header>{item.header}</Accordion.Header>
  //           <Accordion.Body>
  //             <Accordion.Span>{item.body}</Accordion.Span>
  //           </Accordion.Body>
  //         </Accordion.Item>
  //       ))}
  //     </Accordion>,
  //   );

  //   const whatIsNetflixBodyText =
  //     "Netflix is a streaming service that offers a wide variety of award-winning TV programmes, films, anime, documentaries and more – on thousands of internet-connected devices. You can watch as much as you want, whenever you want, without a single advert – all for one low monthly price. There's always something new to discover, and new TV programmes and films are added every week!";

  //   expect(queryByTestId('#faq').prop('style')).toHaveProperty(
  //     'maxHeight',
  //     '0px',
  //   );
  //   fireEvent.click(queryByText('What is Netflix?'));
  //   expect(queryByText(whatIsNetflixBodyText)).toBeTruthy();

  //   fireEvent.click(queryByText('What is Netflix?'));
  //   expect(queryByText(whatIsNetflixBodyText)).toBeFalsy();
  //   // expect(container.firstChild).toMatchSnapshot();
  // });
});
