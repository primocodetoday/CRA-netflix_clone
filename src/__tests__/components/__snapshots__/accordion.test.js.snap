// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> renders the <Accordion /> with populated data 1`] = `
<div
  class="StyledAccordion__Container-k9hf5e-0 jwBtHb"
>
  <div
    class="StyledAccordion__Inner-k9hf5e-1 eWaySA"
  >
    <h1
      class="StyledAccordion__Title-k9hf5e-4 kexgbE"
    >
      Frequently Asked Questions
    </h1>
    <div
      class="StyledAccordion__Item-k9hf5e-2 dDQIVC"
    >
      <div
        class="StyledAccordion__Header-k9hf5e-3 eZdqKN"
      >
        What is Netflix?
        <img
          alt="Open"
          src="/assets/icons/add.png"
        />
      </div>
      <div
        class="StyledAccordion__Body-k9hf5e-5 kNvpUT"
        data-testid="faq"
        style="max-height: 0px;"
      >
        <span
          class="StyledAccordion__Span-k9hf5e-6 bryPYv"
        >
          Netflix is a streaming service that offers a wide variety of award-winning TV programmes, films, anime, documentaries and more – on thousands of internet-connected devices.

You can watch as much as you want, whenever you want, without a single advert – all for one low monthly price. There's always something new to discover, and new TV programmes and films are added every week!
        </span>
      </div>
    </div>
    <div
      class="StyledAccordion__Item-k9hf5e-2 dDQIVC"
    >
      <div
        class="StyledAccordion__Header-k9hf5e-3 eZdqKN"
      >
        How much does Netflix cost?
        <img
          alt="Open"
          src="/assets/icons/add.png"
        />
      </div>
      <div
        class="StyledAccordion__Body-k9hf5e-5 kNvpUT"
        data-testid="faq"
        style="max-height: 0px;"
      >
        <span
          class="StyledAccordion__Span-k9hf5e-6 bryPYv"
        >
          Watch Netflix on your smartphone, tablet, smart TV, laptop or streaming device, all for one low fixed monthly fee. Plans start from £5.99 a month. No extra costs or contracts.
        </span>
      </div>
    </div>
    <div
      class="StyledAccordion__Item-k9hf5e-2 dDQIVC"
    >
      <div
        class="StyledAccordion__Header-k9hf5e-3 eZdqKN"
      >
        Where can I watch?
        <img
          alt="Open"
          src="/assets/icons/add.png"
        />
      </div>
      <div
        class="StyledAccordion__Body-k9hf5e-5 kNvpUT"
        data-testid="faq"
        style="max-height: 0px;"
      >
        <span
          class="StyledAccordion__Span-k9hf5e-6 bryPYv"
        >
          Watch anywhere, anytime, on an unlimited number of devices. Sign in with your Netflix account to watch instantly on the web at netflix.com from your personal computer or on any internet-connected device that offers the Netflix app, including smart TVs, smartphones, tablets, streaming media players and game consoles.

You can also download your favorite programmes with the iOS, Android, or Windows 10 app. Use downloads to watch while you're on the go and without an internet connection. Take Netflix with you anywhere.
        </span>
      </div>
    </div>
    <div
      class="StyledAccordion__Item-k9hf5e-2 dDQIVC"
    >
      <div
        class="StyledAccordion__Header-k9hf5e-3 eZdqKN"
      >
        How do I cancel?
        <img
          alt="Open"
          src="/assets/icons/add.png"
        />
      </div>
      <div
        class="StyledAccordion__Body-k9hf5e-5 kNvpUT"
        data-testid="faq"
        style="max-height: 0px;"
      >
        <span
          class="StyledAccordion__Span-k9hf5e-6 bryPYv"
        >
          Netflix is flexible. There are no annoying contracts and no commitments. You can easily cancel your account online in two clicks. There are no cancellation fees – start or stop your account at any time.
        </span>
      </div>
    </div>
    <div
      class="StyledAccordion__Item-k9hf5e-2 dDQIVC"
    >
      <div
        class="StyledAccordion__Header-k9hf5e-3 eZdqKN"
      >
        What can I watch on Netflix?
        <img
          alt="Open"
          src="/assets/icons/add.png"
        />
      </div>
      <div
        class="StyledAccordion__Body-k9hf5e-5 kNvpUT"
        data-testid="faq"
        style="max-height: 0px;"
      >
        <span
          class="StyledAccordion__Span-k9hf5e-6 bryPYv"
        >
          Netflix has an extensive library of feature films, documentaries, TV programmes, anime, award-winning Netflix originals, and more. Watch as much as you want, any time you want.
        </span>
      </div>
    </div>
  </div>
</div>
`;
